feat: new OCI image index capability. (#347)
kubewarden/docs: docs/writing-policies/spec/host-capabilities/03-container-registry.md

* Input payload: `"busybox:latest"`
* Output payload: `{ "digest": "sha256:69e70a79f2d41ab5d637de98c1e0b055206ba40a8145e7bddb55ccc04e13cf8f"}`

> ## OCI Image index

Reviewer: I would call it `Image manifest`. The Image index is a [type of manifest](https://github.com/opencontainers/image-spec/blob/main/image-index.md):

> The image index is a higher-level manifest which points to specific [image manifests](https://github.com/opencontainers/image-spec/blob/main/manifest.md), ideal for one or more platforms. While the use of an image index is OPTIONAL for image providers, image consumers SHOULD be prepared to process them.

There's no API to fetch the Image index from a registry, that's because you can obtain it by fetching the image manifest.

Because of that, I propose to replace all the `index` instances inside of this PR with `manifest`

Author: Maybe we can change the docs to call it `Image manifest` but making clear that the output can change depending of the OCI artifact manifest requested:

```console
$ crane manifest ghcr.io/kubewarden/policies/pod-privileged:latest | jq ""                                                                                                             
{
  "config": {
    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
    "mediaType": "application/vnd.wasm.config.v1+json",
    "size": 2
  },
  "layers": [
    {
      "digest": "sha256:6d803e1199b0c884741a43c6fbc2f1586839429f00681a3098116437d4210b71",
      "mediaType": "application/vnd.wasm.content.layer.v1+wasm",
      "size": 1285267
    }
  ],
  "schemaVersion": 2
}
```

Therefore, the output can be a `Image index` or a single `Image manifest`. And the policy must deal with it.

Author: @flavio  I've updated the docs changing the name and adding an statement that depending of the image the response can change. It can be a index manifest or a single image manifest.

Reviewer 2: I"m happy with this approach, approving.

> This function fetches OCI images index manifest. This information can
> be used to identify specific images manifests, for one or more platforms.

Reviewer 3: ```suggestion
This function fetches OCI image index manifests. This information can
be used to identify specific image manifests, for one or more platforms.
```

Just a suggestion, I hope I'm not changing the meaning, but I think it reads a little better.

### Caching

> Computing the digest involves a series of network requests between the Kubewarden
> policy host and the remote registry. These operation can be time expensive,
> because of that the results are going to be cached for 1 minute.

Reviewer 3: ```suggestion
Computing the digest involves a series of network requests between the Kubewarden
policy host and the remote registry. These operations can be time expensive so the results are cached for 1 minute.
```

### Authentication

Interactions with private registries require the Kubewarden policy host to
authenticate against the remote registry.

> The policy host will use the same set of credentials used to fetch policies
> from the remote registry.

Reviewer 3: ```suggestion
The policy host uses the same set of credentials as that used to fetch policies
from the remote registry.
```

            "size": 566,
            "annotations": {
                "vnd.docker.reference.digest": "sha256:7d5e84b9314ba7058bfa209881919146ffc4a89c5ba14cfa8270a18c8d418c44",
                "vnd.docker.reference.type": "attestation-manifest"
            },
            "platform": {
                "architecture": "unknown",
                "os": "unknown"
            }
        },
        {
            "mediaType": "application/vnd.oci.image.manifest.v1+json",
            "digest": "sha256:e8904ebb2841dc19ae458436eb01ddea3e8d6ea653c7d9476537f4029b1f45a2",
            "size": 566,
            "annotations": {
                "vnd.docker.reference.digest": "sha256:ae1e98105555f3983496c15c70dafc87639b77830953d6470694b47e0e063e25",
                "vnd.docker.reference.type": "attestation-manifest"
            },
            "platform": {
                "architecture": "unknown",
                "os": "unknown"
            }
        }
    ]
}
```

</td>
</tr>

</table>

For example, when requesting the manifest digest of the `ghcr.io/kubewarden/policy-server:v1.10.0` image,
the payload would be the following ones:

* Input payload: `"ghcr.io/kubewarden/policy-server:v1.10.0"`

> * Output payload: The same of the data used in the table above. Which follows
> the [OCI Image index specification](https://github.com/opencontainers/image-spec/blob/main/image-index.md)

Reviewer: nitpick, the result depends on what the remote image has. It could be a simple manifest, an image index or even a legacy manifest.

I would say something like:

```suggestion
Output payload: the body of the successful response obtained from the registry
```
